### PR TITLE
LPS-167869

### PR DESCRIPTION
**Pull request by alinedoleron**
This PR is related to:
https://issues.liferay.com/browse/LPS-167869
https://issues.liferay.com/browse/LPS-167870

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
Just started reviewing :)

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/Select/SingleSelect.tsx [hunk 1]
```diff
@@ -31,35 +32,69 @@ export function SingleSelect<
...
-					key={index}
-					onClick={() => {
-						setDropdownActive(false);
-						onChange(option);
-					}}
-				>
-					<div>{option.label}</div>
+			{options.map((option, index) => {
+				let events = {};
+				if (option.popover) {
```
Comment (naraandrade) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/Select/SingleSelect.tsx:51]: @alinedoleron We need a blank line between the definition of events and its usage.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx [hunk 2]
```diff
@@ -61,6 +61,9 @@ export default function ActionBuilder({
 	>([]);
 
 	const [objectsOptions, setObjectOptions] = useState<ObjectsOptionsList>([]);
+	const [newObjectActionExecutors, setNewObjectActionExecutors] = useState<
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx:65]: @alinedoleron Could you sort the states by alphabetical order?

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx [hunk 3]
```diff
@@ -154,6 +157,36 @@ export default function ActionBuilder({
...
+					action.disabled = true;
+					action.popover = {
+						body: Liferay.Language.get(
+							'it-is-not-allowed-to-create-update-action-on-after-delete-trigger'
+						),
+						header: Liferay.Language.get('action-not-allowed'),
+					};
+				}
+			});
+			setNewObjectActionExecutors(newObjectActionExecutors);
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx:174]: @alinedoleron Add a blank line before using `newObjectActionExecutors`

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx [hunk 4]
```diff
@@ -154,6 +157,36 @@ export default function ActionBuilder({
...
+			values.objectActionTriggerKey === 'onAfterAdd' ||
+			values.objectActionTriggerKey === 'onAfterUpdate'
+		) {
+			newObjectActionExecutors.map((action) => {
+				if (action.value === 'update-current-entry') {
+					delete action.disabled;
+					delete action.popover;
+				}
+			});
+			setNewObjectActionExecutors(newObjectActionExecutors);
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx:186]: @alinedoleron Same comment as above.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx [hunk 5]
```diff
@@ -315,13 +321,24 @@ export default function ActionBuilder({
 		}));
 	};
 
+	const handleSelectObject = ({
+		target: {value},
+	}: React.ChangeEvent<HTMLSelectElement>) => {
+		const objectDefinitionId = parseInt(value, 10);
+		updateParameters(objectDefinitionId);
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx:328]: @alinedoleron Add a blank line before using `objectDefinitionId`.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx [hunk 6]
```diff
@@ -154,6 +157,36 @@ export default function ActionBuilder({
 		return fields;
 	}, [currentObjectDefinitionFields]);
 
+	useEffect(() => {
+		if (values.objectActionTriggerKey === 'onAfterDelete') {
+			newObjectActionExecutors.map((action) => {
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx:163]: @alinedoleron You need to cover the scenario when the user has select "On After Add" for the trigger (for example) and "Update an Object Entry" for the action, but afterwards they select "On After Delete" for trigger. I believe we should automatically "clean" the action select and return to the state "Choose an Action".

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Pull request closed without merging**